presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -676,18 +677,23 @@ private boolean outerJoinCurrentPosition()
                 return blocked(blocked);
             }
 
-            boolean consumedInput = false;
-            if (needsInput() && inputPage != null) {
+            if (inputPage != null && !consumedInputPage) {
```

Comment:
`inputPage != null` is known to be true

Reply:
> inputPage != null is known to be true

It isn't known. `isFinished()` might return `false` if there is pending `JoinProbe` (e.g because join yielded or output page builder was full).